src/mondo_robot_template.py

```diff
@@ -68,21 +72,23 @@ def run(input_file: str = INPUT_FILE, output_file: str = OUTPUT_FILE, filter_out
 
     # Add additional cols
     out_df['source_medgen_id'] = 'MONDO:MEDGEN'
-    # todo: could optimize by doing apply just on the xref_id col
-    def set_mapping_pred(row):
-        """Set mapping predicate"""
-        # M = Concept UI; D = Descriptor UI; C = SupplementalRecordUI; Q = Qualifier UI
-        pred = 'MONDO:equivalentTo' if row['xref_id'].startswith('mesh:M') \
-            else 'MONDO:relatedTo' if row['xref_id'].startswith('mesh:') \
-            else 'MONDO:equivalentTo'
-        return pred
-    # Context on MeSH IDs:
+    # M = Concept UI; D = Descriptor UI; C = SupplementalRecordUI; Q = Qualifier UI; context:
     #  https://docs.google.com/document/d/1ryu6isBmNEno8lyni70jBaw-D_I6tdEXpnnAK86ZWfs/edit#heading=h.3cho5esard3q
-    out_df['mapping_predicate'] = out_df.apply(set_mapping_pred, axis=1)
+    out_df['mapping_predicate'] = out_df['xref_id'].apply(
+        lambda _id: 'MONDO:equivalentTo' if _id.startswith('mesh:M')
+        else 'MONDO:relatedTo' if _id.startswith('mesh:')
+        else 'MONDO:equivalentTo')
+
+    # Split
+    mesh_selector = out_df['xref_id'].str.startswith('mesh:')
+    out_df_mesh = out_df[mesh_selector]
+    out_df = out_df[~mesh_selector]
 
     # Save
     out_df = pd.concat([pd.DataFrame([ROBOT_ROW_MAP]), out_df])
-    out_df.to_csv(output_file, index=False, sep='\t')
+    out_df.to_csv(outpath_general, index=False, sep='\t')
+    out_df_mesh = pd.concat([pd.DataFrame([ROBOT_ROW_MAP]), out_df_mesh])
+    out_df_mesh.to_csv(outpath_mesh, index=False, sep='\t')
```

Comment:
## Results
[Google Sheet - general (no MeSH)](https://docs.google.com/spreadsheets/d/1OD7WUR-nJ9lUqRv01TTghmf3aGvQRbWV5ZhVhthzB6U/edit#gid=1627855620)
[Google Sheet - MeSH only](https://docs.google.com/spreadsheets/d/1OD7WUR-nJ9lUqRv01TTghmf3aGvQRbWV5ZhVhthzB6U/edit#gid=1949111602)

Reply:
Forget my question in the other PR, now I see it! All good (I only review outputs, not code)